wd_demo.au3

```diff
@@ -204,6 +204,12 @@ Func RunDemo($idDebugging, $idBrowsers, $idUpdate, $idHeadless, $idOutput)
 		EndIf
 
 		ConsoleWrite("+ wd_demo.au3: Running: " & $sDemoName & @CRLF)
+		_WD_CheckContext($sSession, False)
+		If @error Then ; return if session is NOT OK
+			ConsoleWrite("! ---> @error=" & @error & "  @extended=" & @extended & _
+					" : _WD_CheckContext reported a problem with the session" & @CRLF)
+			ContinueLoop
+		EndIf
```

Comment:
Two things wrong here IMO --

- No point in continuing the loop because rest of tests will also fail
- Should use `_RunDemo_ErrorHander` instead of custom code

Reply:
Please take a look: [_WD_CheckContext usage](https://github.com/Danp2/au3WebDriver/pull/360/commits/42aa32cbd335de0b2be0a32120bc20d5a3ffe63b)

Reply:
I still feel there are issues here --

- No need for `$iError = @error` because you don't do anything with the stored error code
- My suggestion was to use `_RunDemo_ErrorHander`, which I believe would handle notifying the user via ConsoleWrite. Instead you simply `ExitLoop` without any type of notice to the user about why the process was aborted.

Please correct me if I'm wrong on the above items.

Reply:
> * No need for `$iError = @error` because you don't do anything with the stored error code

this code 
https://github.com/Danp2/au3WebDriver/blob/42aa32cbd335de0b2be0a32120bc20d5a3ffe63b/wd_demo.au3#L206-L208

in line 207 is setting `$iError` 
and `ExitLoop` from line 208 to line 219 
and then `$iError` is used in line 220
https://github.com/Danp2/au3WebDriver/blob/42aa32cbd335de0b2be0a32120bc20d5a3ffe63b/wd_demo.au3#L219-L222

Do my reasoning is correct ?

Reply:
Yes, you are correct.